src/app/admin/admin.component.ts

```diff
@@ -0,0 +1,121 @@
+
+  constructor(
+    private navbarService: NavbarService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private authService: AuthService,
+    private collectionService: CollectionService,
+    public toaster: ToasterService
+  ) {}
+
+  /**
+   * Returns true if the component is fully initialized (ie the array doesn't contain false), false otherwise
+   *
+   * @readonly
+   * @type {boolean}
+   * @memberof AdminComponent
+   */
+  get initialized(): boolean {
+    return !this._initialized.includes(false);
+  }
+
+  ngOnInit() {
+    // hide CLARK navbar
+    this.navbarService.hide();
+
+    // set the can scroll value to determine whether or not we add 30px of padding to the bottom of the content wrapper
+    const canScroll = this.route.snapshot.firstChild.data.canScroll;
+    this.canScroll = typeof canScroll === 'boolean' ? canScroll : true;
+
+    // listen for route changes to perform the same check above
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      map(() => this.route),
+      map(route => route.firstChild),
+      switchMap(route => route.data),
+    ).subscribe(data =>  {
+      this.canScroll = typeof data.canScroll === 'boolean' ? data.canScroll : true;
+    });
+
+    if (this.authService.hasEditorAccess()) {
+      this.editorMode = true;
+      // we don't need to load collections, so we can set that initialization block to true
+      this._initialized[0] = true;
+    } else {
+      // this user isn't an admin/editor so fetch the route parameter representing the selected collection
+      this.route.paramMap
+        .pipe(takeUntil(this.destroyed$))
+        .subscribe(params => {
+          // we don't need to authorize here since that's done in the route guard
+          this.activeCollection = params.get('collection');
+        });
+
+      this.retrieveAuthorizedCollections().then(() => {
+        // collections array is now set and we should redirect to the first collection
+        if (!this.activeCollection) {
+          this.router.navigate([this.authorizedCollections[0].abvName], { relativeTo: this.route });
+        }
+      }).catch(error => {
+        this.toaster.notify('Error!', 'There was an error retrieving collections. Please try again later.', 'bad', 'far fa-times');
+        console.error(error);
+      });
+    }
+  }
+
+  /**
+   * Retrieve a list of collection names for which the user is authorized to view (has curator status for)
+   *
+   * @memberof AdminComponent
+   */
+  async retrieveAuthorizedCollections() {
+    // wait for user to be logged in (edge case) and then fetch list of curated collections
+    return await this.authService.isLoggedIn.pipe(
+      skipWhile(x => x === false),
+      take(1)
+    )
+    .toPromise()
+    .then(async () => {
+      // we're sure the user is logged in here and so access groups should be defined
+      return await Promise.all(
+        this.authService.user['accessGroups']
```

Comment:
In most cases, a User object in the client doesn't have an `accessGroups` property. This property only exists in the AuthService.

Reply:
So in that case, since the `user` being read is from the `AuthService` where that property exists, shouldn't we have typings for that - allowing `this.authService.user.accessGroups`?

Reply:
Oh actually we do already lol whoops